apps/web/src/client/components/file-uploader/_ui.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * UI features (file list and choose files button)
+ *
+ * @param {Element} uploadForm
+ */
+const registerUIEvents = (uploadForm) => {
+	/** @type {HTMLElement | null} */
+	const uploadButton = uploadForm.querySelector('.pins-file-upload--button');
+	/** @type {HTMLElement | null} */
+	const uploadInput = uploadForm.querySelector('input[name="files"]');
+	/** @type {HTMLElement | null} */
+	const uploadCounter = uploadForm.querySelector('.pins-file-upload--counter');
+	/** @type {HTMLElement | null} */
+	const filesRows = uploadForm.querySelector('.pins-file-upload--files-rows');
```

Comment:
instead of placing the markup for the row in the HTML you could define it in `updateFilesRows` when you generate the item

Reply:
no, I don't think adding HTML in the javascript would be easier to read. It would either treated as a string, something as awful as:

```
const markup = '<li role="listitem" class="pins-file-upload--file-row display--none"><p class="govuk-heading-s" aria-details="File name"></p><button type="button" role="button" class="govuk-link pins-file-upload--remove" aria-details="Remove added file from list">Remove</button></li>'
```

or even worse as an infinite list of attributes as:
```
const newLi = fileListContainer.createNode('li');
newLi.role = 'listirem';
newLi.classList = ...
newLi.ariaLabel = '';
newLi.addChild('button');
newLi.children[0].role = 'button'
etc...
```